From 79aca7271e23fd8e1dd616fb81d3be6d7f3f40ac Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 8 Sep 2014 13:06:42 -0400
Subject: [tools] perf: Fix pmu object compilation error

Message-id: <1410181752-25631-158-git-send-email-jolsa@redhat.com>
Patchwork-id: 91900
O-Subject: [PATCH RHEL7.1 BZ1134356 157/307] perf tools: Fix pmu object compilation error
Bugzilla: 1134356
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Don Zickus <dzickus@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1134356
https://bugzilla.redhat.com/show_bug.cgi?id=1134356

upstream
========
commit 7c2f8164e5415ef48954f6929f4acae5764afdb4
Author: Jiri Olsa <jolsa@redhat.com>
Date: Wed Apr 16 20:49:02 2014 +0200

description
===========
After applying some patches got another shadowing error:

  CC       util/pmu.o
util/pmu.c: In function 'pmu_alias_terms':
util/pmu.c:287:35: error: declaration of 'clone' shadows a global declaration [-Werror=shadow]

Renaming clone to cloned.

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/util/pmu.c b/tools/perf/util/pmu.c
index 00a7dcb..7a811eb 100644
--- a/tools/perf/util/pmu.c
+++ b/tools/perf/util/pmu.c
@@ -284,17 +284,17 @@ static int pmu_aliases(const char *name, struct list_head *head)
 static int pmu_alias_terms(struct perf_pmu_alias *alias,
       struct list_head *terms)
 {
- struct parse_events_term *term, *clone;
+ struct parse_events_term *term, *cloned;
  LIST_HEAD(list);
  int ret;
 
  list_for_each_entry(term, &alias->terms, list) {
-  ret = parse_events_term__clone(&clone, term);
+  ret = parse_events_term__clone(&cloned, term);
   if (ret) {
    parse_events__free_terms(&list);
    return ret;
   }
-  list_add_tail(&clone->list, &list);
+  list_add_tail(&cloned->list, &list);
  }
  list_splice(&list, terms);
  return 0;
-- 
1.7.1